File: ch3/akkademaid-scala/src/main/scala/com/akkademy/askdemo/ParsingActor.scala
package com.akkademy.askdemo

import akka.actor.Actor

class ParsingActor extends Actor{
  override def receive: Receive = {
    case ParseArticle(htmlString) =>
      de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(htmlString)
  }
}

File: ch3/akkademaid-java/src/main/java/com/akkademy/ParsingActor.java
package com.akkademy;


import akka.actor.AbstractActor;
import akka.japi.pf.ReceiveBuilder;
import de.l3s.boilerpipe.extractors.ArticleExtractor;
import scala.PartialFunction;

public class ParsingActor extends AbstractActor {
    public PartialFunction receive() {
        return ReceiveBuilder.
                match(ParseHtmlArticle.class, msg -> {
                    String body = ArticleExtractor.INSTANCE.getText(msg.htmlString);
                    sender().tell(new ArticleBody(msg.uri, body), self());
                }).build();
    }
}
